tools/build/Library/InstallUnpacker/classes/InstallManager.php

```diff
@@ -0,0 +1,182 @@
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+/**
+ * Gateway, manager and DIC for install util classes: download, unzip ...
+ */
+class InstallManager
+{
+    /**
+     * @var ConfigurationValidator
+     */
+    private $validator;
+
+    /**
+     * @var Download
+     */
+    private $download;
+
+    /**
+     * @var Unzip
+     */
+    private $unzip;
+
+    /**
+     * @var string
+     */
+    private $downloadDirectoryPath;
+
+    /**
+     * @var string
+     */
+    private $extractDirectoryPath;
+
+    public function __construct()
+    {
+        $this->validator = new ConfigurationValidator();
+        $this->download = new Download();
+        $this->unzip = new Unzip();
+
+        // @todo: be able to use fallback util directories
+        $this->downloadDirectoryPath = __DIR__ . DIRECTORY_SEPARATOR . 'download';
+        $this->extractDirectoryPath = __DIR__ . DIRECTORY_SEPARATOR . 'extracted';
+    }
+
+    /**
+     * Check whether download of latest PS version can be carried out.
+     *
+     * @return string[] array of issues. Empty if download can be performed.
+     */
+    public function testDownloadCapabilities()
+    {
+        return $this->validator->testSystemCanPerformDownloadUnzipAndReplace();
+    }
+
+    /**
+     * @return string
+     *
+     * @throws \RuntimeException
+     */
+    public function getLatestStableAvailableVersion()
+    {
+        return $this->download->getLatestStableAvailableVersion();
+    }
+
+    /**
+     * @return bool
+     *
+     * @throws PrestashopCouldNotInstallLatestVersionException
+     */
+    public function downloadUnzipAndReplaceLatestPSVersion()
+    {
+        if (is_dir($this->downloadDirectoryPath) || file_exists($this->downloadDirectoryPath)) {
+            throw new PrestashopCouldNotInstallLatestVersionException(sprintf('Directory %s already exists.', $this->downloadDirectoryPath));
+        }
+
+        // create 'download' directory
+        $createDirectoryResult = @mkdir($this->downloadDirectoryPath);
+        if (false === $createDirectoryResult) {
+            throw new PrestashopCouldNotInstallLatestVersionException(sprintf('Could not create directory %s', $this->downloadDirectoryPath));
```

Comment:
Line too long

Reply:
if php cs fixer is not complaining, we shouldn't do this kind of comments, don't we?

Reply:
We need to, otherwise you can have a line with 30000 characters :/ 
As I said before, Php Cs fixer don't fix all PSR2 errors and warnings. 
More it's hard to review a line you can't read easily :/

Reply:
we could add https://github.com/squizlabs/PHP_CodeSniffer too, but the main idea is to avoid losing time on aesthetics when doing review.

Reply:
He said me he's agreed with the line too long, just miss the change with the lot of rebase he needed to do recently :D

Reply:
it's not about agreement ;) it about keeps your brain free and available for tasks that require a brain :D :D

Reply:
It takes me time and my brain to use my mouse to scroll to left / right to views what are hidden due to long line :(